src/logics/LIALogic.cc

```diff
@@ -162,4 +162,90 @@ const SymRef LIALogic::get_sym_Num_EQ () const {return sym_Int_EQ;}
 const SymRef LIALogic::get_sym_Num_ZERO () const {return sym_Int_ZERO;}
 const SymRef LIALogic::get_sym_Num_ONE () const {return sym_Int_ONE;}
 const SymRef LIALogic::get_sym_Num_ITE () const {return sym_Int_ITE;}
-const SRef LIALogic::get_sort_NUM () const {return sort_INTEGER;}
\ No newline at end of file
+const SRef LIALogic::get_sort_NUM () const {return sort_INTEGER;}
+
+PTRef LIALogic::sumToNormalizedInequality(PTRef sum) {
+    vec<PTRef> varFactors;
+    PTRef constant = PTRef_Undef;
+    Pterm const & s = getPterm(sum);
+    for (int i = 0; i < s.size(); i++) {
+        if (isConstant(s[i])) {
+            assert(constant == PTRef_Undef);
+            constant = s[i];
+        } else {
+            assert(isLinearFactor(s[i]));
+            varFactors.push(s[i]);
+        }
+    }
+    if (constant == PTRef_Undef) { constant = getTerm_NumZero(); }
+    auto constantValue = getNumConst(constant);
+    termSort(varFactors);
+    vec<PTRef> vars; vars.capacity(varFactors.size());
```

Comment:
vec\<PTRef\> vars(varFactors.size());

Reply:
If I do that, then it will default-construct the given number of elements and then in the loop I cannot use `push` but I have to re-assign the elements, correct? Is that better then reserving just the capacity and then pushing the elements?

Reply:
I think it's good like you proposed.  Ignore that one.